notification/notification.go

```diff
@@ -0,0 +1,93 @@
+package notification
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/go-redis/redis/v7"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+)
+
+const (
+	EnvRedisHost = "REDIS_HOST"
+	EnvRedisPort = "REDIS_PORT"
+)
+
+var DefaultRedisOpts = redis.Options{
+	MinIdleConns:       2,               // allow the pool to downsize to 0 conns.
+	PoolSize:           5,               // one for subscriptions, one for nonblocking operations.
+	PoolTimeout:        3 * time.Minute, // amount of time a waiter will wait for a conn to become available.
+	MaxRetries:         30,
+	MinRetryBackoff:    1 * time.Second,
+	MaxRetryBackoff:    3 * time.Second,
+	DialTimeout:        10 * time.Second,
+	ReadTimeout:        10 * time.Second,
+	WriteTimeout:       10 * time.Second,
+	IdleCheckFrequency: 30 * time.Second,
+	MaxConnAge:         2 * time.Minute,
+}
+
+var client *redis.Client
+
+func init() {
+	var (
+		port = 6379
+		host = os.Getenv(EnvRedisHost)
+	)
+
+	if portStr := os.Getenv(EnvRedisPort); portStr != "" {
+		var err error
+		port, err = strconv.Atoi(portStr)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	opts := DefaultRedisOpts
+	opts.Addr = fmt.Sprintf("%s:%d", host, port)
+
+	client = redis.NewClient(&opts)
+}
+
+func Fetch(rp *runtime.RunParams, lastId string) ([]*runtime.Notification, string, error) {
```

Comment:
Shouldn't this take a context and return a channel?

Reply:
We probably want to consume notifications as they are happening.

Reply:
The use-case right now is to fetch the `outcome` of the test plans at the end of a test run. I agree that the signature can be improved though (this logic stays on Testground daemon at the moment).

We don't want to consume notifications as they are happening, at least not now. We will want to do that in the future.